flow/raster_cache.cc

```diff
@@ -175,25 +176,63 @@ std::unique_ptr<RasterCacheResult> RasterCache::RasterizeDisplayList(
 
 void RasterCache::Prepare(PrerollContext* context,
                           Layer* layer,
-                          const SkMatrix& ctm) {
-  RasterCacheKey cache_key(layer->unique_id(), RasterCacheKeyType::kLayer, ctm);
-  Entry& entry = cache_[cache_key];
+                          const SkMatrix& ctm,
+                          RasterCacheLayerStrategy strategy) {
+  auto cache_key_optional =
+      TryToMakeRasterCacheKeyForLayer(layer, strategy, ctm);
+  if (!cache_key_optional) {
+    return;
+  }
+  Entry& entry = cache_[cache_key_optional.value()];
   entry.access_count++;
   entry.used_this_frame = true;
   if (!entry.image) {
-    entry.image = RasterizeLayer(context, layer, ctm, checkerboard_images_);
+    entry.image =
+        RasterizeLayer(context, layer, strategy, ctm, checkerboard_images_);
+  }
+}
+
+std::optional<RasterCacheKey> RasterCache::TryToMakeRasterCacheKeyForLayer(
+    const Layer* layer,
+    RasterCacheLayerStrategy strategy,
+    const SkMatrix& ctm) const {
+  switch (strategy) {
+    case RasterCacheLayerStrategy::kLayer:
+      return RasterCacheKey(layer->unique_id(), RasterCacheKeyType::kLayer,
+                            ctm);
+    case RasterCacheLayerStrategy::kLayerChildren:
+      FML_DCHECK(layer->as_container_layer());
+      auto& children_layers = layer->as_container_layer()->layers();
+      auto children_count = children_layers.size();
+      if (children_count == 0) {
+        return std::nullopt;
+      } else if (children_count == 1) {
```

Comment:
Adding this new case, won't the parent be painted in Draw instead of the single child? The Draw call is called with the parent as the "layer" and the action taken for kLayer is to paint that layer - but this will paint the parent layer which applies some sort of filter, not the child without the filter.

Reply:
I think it is addressed since the code was removed.

Actually, in the previous implementation, there is no need to worry here. Because here is the logic for the key generation only. when rasterizing layer, it is distinguished according to the `RasterCacheLayerStrategy`.  So eventually the branch of layer children will be executed.

``` c++
switch (strategy) {
  case RasterCacheLayerStrategy::kLayer:
    if (layer->needs_painting(paintContext)) {
      layer->Paint(paintContext);
    }
    break;
  case RasterCacheLayerStrategy::kLayerChildren:
    FML_DCHECK(layer->as_container_layer());
    layer->as_container_layer()->PaintChildren(paintContext);
    break;
}
```